xrayvision/visibility.py

```diff
@@ -406,6 +438,53 @@ def instrument(self):
             i += 1
         return instr
 
+    @property
+    def __eq__(self, other):
+        if not _attrs_both_none_or_neither(self.observer_coordinate, other.observer_coordinate):
+            return False
+        if hasattr(self.observer_coordinate, "__len__"):
+            if not hasattr(other.observer_coordinate, "__len__"):
```

Comment:
I'm not sure I follow this if they both have a `__len__` should compare them and if they don't they are not equal? No if only one has `__len__`?

Reply:
Yep, this looks wrong.  I'll suggest a fix.